lizard.py

```diff
@@ -759,8 +766,14 @@ def get_warnings(code_infos, option):
 
 def print_result(result, option, scheme):
     result = print_and_save_modules(result, option.extensions, scheme)
+    warning_count = 0
+    warning_nloc = 0
     warnings = get_warnings(result, option)
-    warning_count, warning_nloc = print_warnings(option, scheme, warnings)
+    if list(warnings):
+        warnings = get_warnings(result, option)
```

Comment:
Perhaps get_generator_of_warnings?

> On 5 Sep 2016, at 10:46 PM, Olzhas Rakhimov <notifications@github.com> wrote:
> 
> In lizard.py <https://github.com/terryyin/lizard/pull/151#discussion_r77532042>:
> 
> >      warnings = get_warnings(result, option)
> > -    warning_count, warning_nloc = print_warnings(option, scheme, warnings)
> > +    if list(warnings):
> > +        warnings = get_warnings(result, option)
> @terryyin <https://github.com/terryyin> some documentation should be enough; otherwise, maybe generate_warnings would be a better alternative.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/terryyin/lizard/pull/151/files/016d0ac2dbabf05e808165c6d8778dc922821097#r77532042>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAwJYrJ5cdfil00BKRjnlyBisB2zop0fks5qnCtbgaJpZM4J0DSV>.
>